src/tools/bin/plot_chargeConservation.py

```diff
@@ -0,0 +1,210 @@
+    """
+    sets label and font size of color bar
+
+    Parameters:
+    cb: pyplot colorbar object
+        colorbar to be adjusted
+    """
+    cb.set_label(r"$\vec \nabla \vec E - \rho / \varepsilon_0"+
+                 r" \,[ \mathrm{max}(|\rho_k|) / \varepsilon_0 ]$",
+                 fontsize=20)
+    for t in cb.ax.get_yticklabels():
+        t.set_fontsize(14)
+
+def plotError(h5file, slice_pos=[0.5, 0.5, 0.5]):
+    """
+    read field data from hdf5 files
+    compute div(E) - rho/epsilon_0
+    plot slices through simulation volume
+
+    Parameters:
+    h5file: file name 
+        file name to hdf5 data set from PIConGPU
+            
+    slice_pos: list of floats
+        list of 3 floats to define slice position [0, 1]
+        Default=[0.5, 0.5, 0.5]
+    """
+    # load hdf5 file
+    f = h5py.File(h5file, "r")
+
+    # read time step (python 2 and 3 save)
+    timestep = -1
+    for i in f['/data'].keys():
+        timestep = i
+
+    # load physics constants and simulation parameters
+    EPS0 = f["/data/{}".format(timestep)].attrs["eps0"]
+    CELL_WIDTH = f["/data/{}".format(timestep)].attrs["cell_width"]
+    CELL_HEIGHT = f["/data/{}".format(timestep)].attrs["cell_height"]
+    CELL_DEPTH = f["/data/{}".format(timestep)].attrs["cell_depth"]
+    CELL_VOLUME = CELL_WIDTH * CELL_HEIGHT * CELL_DEPTH
+
+    # load electric field
+    Ex = np.array(f["/data/{}/fields/FieldE/x".format(timestep)])
+    Ey = np.array(f["/data/{}/fields/FieldE/y".format(timestep)])
+    Ez = np.array(f["/data/{}/fields/FieldE/z".format(timestep)])
+
+    # load and add charge density
+    charge = np.zeros_like(Ex)
+    norm = 0.0
+    for field_name in f["/data/{}/fields/".format(timestep)].keys():
+        if field_name[0:8] == "Density_":
+            # load species density
+            species_Density = np.array(f["/data/{}/fields/".format(timestep) + field_name])
+            # choose norm to be the maximal charge density of all species
+            norm = np.max([norm, np.amax(np.abs(species_Density))])
+            # add charge density to total charge density
+            charge += species_Density
+
+    # close hdf5 file
+    f.close()
+
+    # compute divergence of electric field according to Yee lattice
+    div = ((Ex[1:, 1:, 1:] - Ex[1:, 1:, :-1])/CELL_WIDTH + 
+           (Ey[1:, 1:, 1:] - Ey[1:, :-1, 1:])/CELL_HEIGHT + 
+           (Ez[1:, 1:, 1:] - Ez[:-1, 1:, 1:])/CELL_DEPTH)
+
+    # compute difference between electric field divergence and charge density
+    diff = (div  - charge[1:, 1:, 1:]/EPS0)
+
+    limit = np.amax(np.abs(diff))
+
+    # plot result
+    plt.figure(figsize=(14, 5))
+
+    plt.subplot(131)
+    slice_cell_z = np.int(np.floor((diff.shape[0]-1)*slice_pos[0]))
+    plt.title("slice in z at {}".format(slice_cell_z), fontsize=20)    
+    plt.imshow(diff[slice_cell_z, :, :], 
+               vmin=-limit, vmax=+limit,
+               aspect='auto')
+    plt.xlabel("$x\,[\Delta x]$", fontsize=20)
```

Comment:
Each axis has a different length (x=256, y=512, z=128). Since the ticks show the length in cells, the correctness can be checked without interpreting the color.